pkg/operation/botanist/component/dependencywatchdog/bootstrap.go

```diff
@@ -331,19 +328,33 @@ func (b *bootstrapper) getLabels() map[string]string {
 	return map[string]string{v1beta1constants.LabelApp: b.name()}
 }
 
-func (b *bootstrapper) clusterRoleRules() []rbacv1.PolicyRule {
+func (b *bootstrapper) roleRules() []rbacv1.PolicyRule {
 	switch b.values.Role {
-	case RoleEndpoint:
+	case RoleWeed:
 		return []rbacv1.PolicyRule{
 			{
 				APIGroups: []string{""},
-				Resources: []string{"endpoints"},
-				Verbs:     []string{"get", "list", "watch"},
+				Resources: []string{"endpoints", "events"},
+				Verbs:     []string{"create", "get", "update", "patch"},
+			},
+			{
+				APIGroups: []string{"coordination.k8s.io"},
+				Resources: []string{"leases"},
+				Verbs:     []string{"create"},
 			},
+			{
+				APIGroups:     []string{"coordination.k8s.io"},
+				ResourceNames: []string{"dependency-watchdog-weed"},
+				Resources:     []string{"leases"},
+				Verbs:         []string{"get", "watch", "update"},
+			},
+		}
+	case RoleProbe:
+		return []rbacv1.PolicyRule{
 			{
 				APIGroups: []string{""},
-				Resources: []string{"pods"},
-				Verbs:     []string{"get", "list", "watch", "delete"},
+				Resources: []string{"events"},
```

Comment:
do we need this?

Reply:
i am not sure, I'll check. It was already there so I didn't change , but I also feel its not needed

Reply:
I checked without giving permissions for `events` but then we get following error:
```
2023-01-18T12:27:35Z	INFO	dwd.endpoints-controller	Starting Controller	{"controller": "endpoints", "controllerGroup": "", "controllerKind": "Endpoints"}
E0118 12:27:35.589634       1 event.go:267] Server rejected event '&v1.Event{TypeMeta:v1.TypeMeta{Kind:"", APIVersion:""}, ObjectMeta:v1.ObjectMeta{Name:"dwd-weeder-leader-election.173b66ed3b57ed41", GenerateName:"", Namespace:"garden", SelfLink:"", UID:"", ResourceVersion:"", Generation:0, CreationTimestamp:time.Date(1, time.January, 1, 0, 0, 0, 0, time.UTC), DeletionTimestamp:<nil>, DeletionGracePeriodSeconds:(*int64)(nil), Labels:map[string]string(nil), Annotations:map[string]string(nil), OwnerReferences:[]v1.OwnerReference(nil), Finalizers:[]string(nil), ManagedFields:[]v1.ManagedFieldsEntry(nil)}, InvolvedObject:v1.ObjectReference{Kind:"Lease", Namespace:"garden", Name:"dwd-weeder-leader-election", UID:"981a3a4b-1d86-4701-9eb0-c3abcda023f8", APIVersion:"coordination.k8s.io/v1", ResourceVersion:"23263", FieldPath:""}, Reason:"LeaderElection", Message:"dependency-watchdog-weeder-5596797b8c-658bl_a6dd3e24-e94d-41d3-b6c9-b2941386bdb8 became leader", Source:v1.EventSource{Component:"dependency-watchdog-weeder-5596797b8c-658bl_a6dd3e24-e94d-41d3-b6c9-b2941386bdb8", Host:""}, FirstTimestamp:time.Date(2023, time.January, 18, 12, 27, 35, 587171649, time.Local), LastTimestamp:time.Date(2023, time.January, 18, 12, 27, 35, 587171649, time.Local), Count:1, Type:"Normal", EventTime:time.Date(1, time.January, 1, 0, 0, 0, 0, time.UTC), Series:(*v1.EventSeries)(nil), Action:"", Related:(*v1.ObjectReference)(nil), ReportingController:"", ReportingInstance:""}': 'events is forbidden: User "system:serviceaccount:garden:dependency-watchdog-weeder" cannot create resource "events" in API group "" in the namespace "garden"' (will not retry!)
```

The error says that the controller on acquiring leader-election is trying to create an event but it fails because of lack of priviledges. I think the leader-election code creates an event everytime something related to leader-election occurs.
So I will revert the change of removing `events` priviledges.

Reply:
After reverting the changes its successful , the events look like 
```
69s         Normal    LeaderElection      lease/dwd-prober-leader-election                      dependency-watchdog-prober-bd5666d4b-2hm85_c1b18ea1-c4a1-48f1-b141-0d3b67a89a7d became leader
59s         Normal    LeaderElection      lease/dwd-weeder-leader-election                      dependency-watchdog-weeder-5596797b8c-qg659_6435da5a-c7dd-49cc-8ea3-c6b25252753b became leader
```